ext/oauth-client/CRM/OAuth/Upgrader.php

```diff
@@ -36,4 +36,16 @@ public function upgrade_0001(): bool {
     return TRUE;
   }
 
+  /**
+   * Add support for tenancy
+   *
+   * @return bool
+   * @throws Exception
+   */
+  public function upgrade_5581(): bool {
+    $this->ctx->log->info('Applying oauth-client update 5581. Adding tenant column to the civicrm_oauth_client table.');
+    $this->addTask('Add Tenant ID to civicrm_oauth_client', 'addColumn', 'civicrm_oauth_client', 'tenant', 'varchar(128) NULL COMMENT "Tenant ID" AFTER guid');
```

Comment:
Please can you wrap this line in a check so it's re-runnable:
```suggestion
  if (!CRM_Core_BAO_SchemaHandler::checkIfFieldExists('civicrm_oauth_client', 'tenant')) {
    $this->addTask('Add Tenant ID to civicrm_oauth_client', 'addColumn', 'civicrm_oauth_client', 'tenant', 'varchar(128) NULL COMMENT "Tenant ID" AFTER guid');
  }
```

Reply:
@mattwire thanks for the eyes on this. I'm pretty sure the 'addColumn' task is already wrapped in the CRM_Core_BAO_SchemaHandler::checkIfFieldExists, if you look at: \CRM_Upgrade_Incremental_Base::addColumn [=

Reply:
Oh yes you are right!